pkg/k8s/api/outdated.go

```diff
@@ -0,0 +1,93 @@
+package api
+
+import (
+	"encoding/json"
+	bolt "go.etcd.io/bbolt"
+	"golang.org/x/xerrors"
+	"io"
+	"path/filepath"
+
+	"github.com/aquasecurity/trivy-db/pkg/db"
+	"github.com/aquasecurity/trivy-db/pkg/types"
+	"github.com/aquasecurity/trivy-db/pkg/utils"
+)
+
+const (
+	k8sOutdatedAPiDir                = "api"
+	K8sOutdatedAPI    types.SourceID = "outdated-api"
+)
+
+type Advisory []types.OutDatedAPIData
+
+var (
+	dataSource = "outdatedapi-data-source"
+	dataType   = "outdated-api"
+
+	source = types.DataSource{
+		ID:   K8sOutdatedAPI,
+		Name: "Kubernetes GitHub docs",
+		URL:  "https://github/kubernetes/kubernetes",
+	}
+)
+
+type Outdated struct {
+	dbc db.Operation
+}
+
+func NewOutdated() Outdated {
+	return Outdated{
+		dbc: db.Config{},
+	}
+}
+
+func (vs Outdated) Name() types.SourceID {
+	return source.ID
+}
+
+func (vs Outdated) Update(dir string) error {
+	rootDir := filepath.Join(dir, "trivy-db-data", "k8s", k8sOutdatedAPiDir)
+	var advisories []Advisory
+	err := utils.FileWalk(rootDir, func(r io.Reader, path string) error {
+		var advisory Advisory
+		if err := json.NewDecoder(r).Decode(&advisory); err != nil {
+			return xerrors.Errorf("failed to decode k8s outdated api Advisory: %w", err)
+		}
+		advisories = append(advisories, advisory)
+		return nil
+	})
+	if err != nil {
+		return xerrors.Errorf("error in outdated api walk: %w", err)
+	}
+
+	if err = vs.save(advisories); err != nil {
+		return xerrors.Errorf("error in outdated api save: %w", err)
+	}
+
+	return nil
+}
+
+func (vs Outdated) save(advisories []Advisory) error {
+	err := vs.dbc.BatchUpdate(func(tx *bolt.Tx) error {
+		if err := vs.dbc.PutK8sDataSource(tx, dataSource, source); err != nil {
```

Comment:
i am not sure, but, perhaps, we need to save `DataSourse` with all `Datasourses`?
I was confused when i didn't find `DataSourse` in default bucket.

Reply:
we are trying to separate vulnerabilities sources from other sources , as we might have different sources type in the future

Reply:
Okay. It makes sense.
But maybe we need to separate `data sources` and `databases`. When we add several new sources, confusion can arise because `data sources` and `databases` will be saved in 1 bucket.
e.g.
```yaml
k8s:
    - outdated-api
    - outdatedapi-data-source
    - new-db
    - new-db-data-source
    - new-db-2
    - new-db-2-data-source
    ...
```
What if we create 2 buckets inside `k8s` for `data sources` and `db's`
like this:
```yaml
k8s:
    - db(need to choose the best name)
        - outdated-api
        - new-db
        - new-db-2
    - data-sources
        - outdatedapi-data-source
        - new-db-data-source
        - new-db-2-data-source
    ...
```